revolv/project/views.py

```diff
@@ -62,15 +77,28 @@ def get_success_url(self):
         messages.success(self.request, 'Project details updated')
         return reverse('project:view', kwargs={'pk': self.get_object().id})
 
+    # validates project, formset of donation levels, and adds categories as well
     def form_valid(self, form):
-        project = self.get_object()
-        project.update_categories(form.cleaned_data['categories_select'])
+        context = self.get_context_data()
```

Comment:
I think helper method is the way to go
On Fri, Mar 20, 2015 at 11:35 PM Eric Shen <notifications@github.com> wrote:

> In revolv/project/views.py
> <https://github.com/calblueprint/revolv/pull/204#discussion_r26889714>:
>
> >      def form_valid(self, form):
> > -        project = self.get_object()
> > -        project.update_categories(form.cleaned_data['categories_select'])
> > +        context = self.get_context_data()
>
> Will look into this mid break (like Wednesday-ish). Unfortunately, there
> might not be a way to do this without calling get_context_data (though I
> can define a helper method)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/calblueprint/revolv/pull/204/files#r26889714>.
>